libs/components/filter-bar/src/lib/modules/filter-bar/filter-item-modal.component.ts

```diff
@@ -34,13 +39,39 @@ import { SkyFilterItem } from './models/filter-item';
   ],
 })
 export class SkyFilterItemModalComponent implements SkyFilterItem {
+  /**
+   * A unique identifier for the filter item.
+   * @required
+   */
   public readonly filterId = input.required<string>();
+
+  /**
+   * The label text for the filter item.
+   * @required
+   */
   public readonly labelText = input.required<string>();
+
   /**
-   * The configuration options for showing a custom filter modal.
+   * The modal component to display when the user selects the filter.
+   * The component needs to inject a `SkyFilterBarFilterModalContext` object on instantiation to receive context from the modal service.
+   * The return value of the modal save action needs to be a `SkyFilterBarFilterValue`.
+   * @required
    */
-  public readonly filterModalConfig =
-    input.required<SkyFilterBarFilterModalConfig>();
+  public readonly modalComponent = input.required<Type<unknown>>();
+
+  /**
+   * The size of the modal to display. The valid options are `'small'`, `'medium'`, `'large'`, and `'fullScreen'`.
+   * @default 'medium'
+   */
+  public readonly modalSize = input<SkyFilterBarFilterModalSizeType>();
+
+  /**
+   * Fires when the user clicks on the filter item. Consumers passing additional context data to a filter modal
```

Comment:
```suggestion
   * Fires when the user clicks the filter item. To pass additional context data to a filter modal, consumers
```
In the first sentence, I'm just removing an extra word. In the second sentence, I rearranged to use an opening clause because I think it makes this easier to follow. Both are very minor tweaks, so please disregard either if you disagree.